src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9863,8 +9870,30 @@ private StatementSyntax ParseLocalDeclarationStatement(SyntaxList<AttributeListS
                     {
                         mods[i] = this.AddError(mod, ErrorCode.ERR_BadMemberFlag, mod.Text);
                     }
+                    else if (mod.Kind == SyntaxKind.ScopedKeyword)
+                    {
+                        // Make 'scoped' part of the type when it is the last token in the modifiers list
```

Comment:
>Make 'scoped' part of the type when it is the last token in the modifiers list [](http://example.com/codeflow?start=27&length=78)

Is this different than handling of `ref`? In short, could treat the first `scoped` keyword as part of the type and create a `ScopedTypeSyntax` at that point?

Reply:
> Is this different than handling of ref? 

```ParseType``` is called in many places and I prefer to control when we actually can create ```ScopedType``` since we don't want to create it everywhere.

> could treat the first scoped keyword as part of the type and create a ```ScopedTypeSyntax``` at that point?

I didn't want to diverge the logic from other places where we treat ```scoped``` as a true modifier and prefer ```ParseDeclarationModifiers``` to consume the token. Given that, it is much easier to move the modifier than to reset type parsing.

Reply:
> I didn't want to diverge the logic from other places where we treat scoped as a true modifier and prefer ```ParseDeclarationModifiers``` to consume the token. 

To clarify, the goal of this PR is to change the representation that parser produces. I did not mean to adjust the rules (or their implementation) that we use to decide whether "scoped" is a keyword or an identifier. To put it in other words, if we used to create a node with modifiers followed by ```VariableDeclarationSyntax``` and the last modifier is ```scoped```,  change the node that we create to use ```ScopedTypeSyntax``` inside ```VariableDeclarationSyntax``` instead. Implementation does exactly that. 

>>could treat the first scoped keyword as part of the type

I guess we could. I am also aware of a decision made at LDM ([https://github.com/dotnet/csharplang/blob/main/meetings/2022/LDM-2022-07-27.md](https://github.com/dotnet/csharplang/blob/main/meetings/2022/LDM-2022-07-27.md)): "We'd like to think of this as a new kind of modifier, a modifier that must go directly on a type, rather than a modifier that can go anywhere in the modifiers list of a method." Implementing this view proper requires changing how we parse modifiers in general in all contexts. For example, We probably want to stop treating "scoped" as a modifier for the purpose of parsing. I.e. modifiers stop at "scoped" and do not consume it. In syntax model for parameters (and other similar places), we will put ```scoped``` to the modifiers list at the end. A change like that is a quite big change and I was planning to do that separately. It is quite possible that after that change this function could be simplified and the ```scoped``` could be handled upfront.